### PR TITLE
feat(as-4161): full Planning Dev & Unit test for Granted or Refused screen

### PR DESCRIPTION
**Pull request by nssdesikan**
## Ticket Number
https://pins-ds.atlassian.net/browse/AS-4161

## Description of change
full Planning Dev & Unit test for Granted or Refused screen

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [x] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by demzzydev — CHANGES_REQUESTED** [review]
This branch needs to shake off some commits as it seems some of Team 1's work have made it here.

File: packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      expect(rule.locations).toEqual(['body']);
+      expect(rule.optional).toBeFalsy();
+      expect(rule.stack).toHaveLength(3);
+
+      expect(rule.stack[0].negated).toBeTruthy();
+      expect(rule.stack[0].validator.name).toEqual('isEmpty');
+      expect(rule.stack[0].message).toEqual(
+        'Select if your planning application was granted or refused, or if you have not received a decision'
+      );
+
```
Comment (demzzydev) [inline packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js:29]: Not sure if intentional but it seems the an entry has been skipped within the stack array.

File: packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          expect(result.errors[0].location).toEqual('body');
+          expect(result.errors[0].msg).toEqual(
+            'Select if your planning application was granted or refused, or if you have not received a decision'
+          );
+          expect(result.errors[0].param).toEqual('granted-or-refused');
+          expect(result.errors[0].value).toEqual(undefined);
+        },
+      },
+      {
+        title: 'User selected `refused` on the full planning applicaxtion question',
```
Comment (demzzydev) [inline packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js:54]: Typo here

Comment (nssdesikan) [reply packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js:54]: Corrected

File: packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js (hunk 1 above)
Comment (nssdesikan) [reply packages/forms-web-app/__tests__/unit/validators/full-planning/granted-or-refused.test.js:29]: Stack[1] is nothing but bail. So its intentional.

**Review by demzzydev — APPROVED** [review]
LGTM

**Pull request merged**